ballerina/http_client_payload_builder.bal

```diff
@@ -159,18 +159,30 @@ isolated function jsonPayloadBuilder(Response response, TargetType targetType) r
     }
 }
 
-isolated function nonNilablejsonPayloadBuilder(Response response, typedesc<anydata> targetType)
+isolated function nonNilablejsonPayloadBuilder(Response response, typedesc<anydata> targetType, boolean requireLaxDataBinding)
         returns anydata|ClientError {
     json payload = check response.getJsonPayload();
-    var result = jsondata:parseAsType(payload, {enableConstraintValidation: false, allowDataProjection: false}, targetType);
+    var result = jsondata:parseAsType(
+        payload,
+        requireLaxDataBinding 
```

Comment:
Shall we move the options outside and define a variable for that:
```bal
jsondata:Options jsonParserOptions = {
    enableConstraintValidation: false,
    allowDataProjection: requireLaxDataBinding ? {nilAsOptionalField: true, absentAsNilableType: true} : false
};
```

I think this more readable